File: appMain/src/main/java/com/cartlc/tracker/etc/BootstrapData.java
package com.cartlc.tracker.etc;

import com.cartlc.tracker.app.TBApplication;
import com.cartlc.tracker.data.DataAddress;
import com.cartlc.tracker.data.DataNote;
import com.cartlc.tracker.data.TableAddress;
import com.cartlc.tracker.data.TableCollectionEquipmentProject;
import com.cartlc.tracker.data.TableCollectionNoteProject;
import com.cartlc.tracker.data.TableProjects;

import java.util.Arrays;

/**
 * Created by dug on 5/9/17.
 */

public class BootstrapData {

    public static void Init() {
        if (TableAddress.getInstance().count() == 0) {
            TableAddress.getInstance().add(Arrays.asList(ADDRESSES));
        }
        if (TableProjects.getInstance().count() == 0) {
            AddCollections();
            AddNotes();
        }
    }

    static final DataAddress[] ADDRESSES = {
            new DataAddress("Alamo Concrete"),
            new DataAddress("CW Roberts"),
            new DataAddress("Central Concrete"),
            new DataAddress("IMI"),
            new DataAddress("Ozinga"),
            new DataAddress("Wingra Ready Mix"),
            new DataAddress("Point Ready Mix"),
    };

    static void AddCollections() {
        TableCollectionEquipmentProject.getInstance().addByName("Five Cubits", Arrays.asList(new String[]{
                "OBC",
                "OBC Bracket",
                "RDT",
                "VMX",
                "Tablet",
                "Charging Converter",
                "6 pin Canbus Cable",
                "9 pin Canbus Cable",
                "Green Canbus Cable",
                "ODB II Canbus Cable",
                "GDS Cup",
                "External Antenna",
                "Internal Antenna",
                "External Speaker",
                "Microphone",
                "Tablet",
                "Repair Work",
                "Speaker Box",
                "Other",
        }));
        TableCollectionEquipmentProject.getInstance().addByName("Digital Fleet", Arrays.asList(new String[]{
                "Antenna",
                "Tablet",
                "Modem",
                "JBox",
                "Canbus",
                "Ram Mount/Cradle",
                "Charging Converter",
                "Repair Work",
                "Uninstall",
                "Other",
        }));
        TableCollectionEquipmentProject.getInstance().addByName("Smart Witness", Arrays.asList(new String[]{
                "KP1S",
                "CPI",
                "SVC 1080",
                "Modem",
                "Driver Facing Camera",
                "Back Up Camera",
                "Side Camera 1",
                "Side Camera 2",
                "Other",
        }));
        TableCollectionEquipmentProject.getInstance().addByName("Fed Ex", Arrays.asList(new String[]{
                "KP1S",
                "SVA30",
                "Modem",
                "Mobileye",
                "Backup Sensors",
                "Other",
        }));
        TableCollectionEquipmentProject.getInstance().addByName("Verifi", Arrays.asList(new String[]{
                "Other",
        }));
        TableProjects.getInstance().addTest(TBApplication.OTHER);
    }

    static void AddNotes() {
        TableCollectionNoteProject.getInstance().addByName("Five Cubits", Arrays.asList(new DataNote[]{
                new DataNote("Serial #", DataNote.Type.ALPHANUMERIC),
                new DataNote("IMEI #", DataNote.Type.NUMERIC),
                new DataNote("Other", DataNote.Type.MULTILINE),
        }));
        TableCollectionNoteProject.getInstance().addByName("Digital Fleet", Arrays.asList(new DataNote[]{
                new DataNote("Serial #"),
                new DataNote("IMEI #"),
                new DataNote("Other"),
        }));
        TableCollectionNoteProject.getInstance().addByName("Smart Witness", Arrays.asList(new DataNote[]{
                new DataNote("Serial #"),
                new DataNote("IMEI #"),
                new DataNote("Sim #", DataNote.Type.NUMERIC_WITH_SPACES),
                new DataNote("DRID #", DataNote.Type.TEXT),
                new DataNote("Other"),
        }));
        TableCollectionNoteProject.getInstance().addByName("Fed Ex", Arrays.asList(new DataNote[]{
                new DataNote("Serial #"),
                new DataNote("IMEI #"),
                new DataNote("Sim #"),
                new DataNote("DRID #"),
                new DataNote("Mobileye", DataNote.Type.TEXT),
                new DataNote("Other"),
        }));
        TableCollectionNoteProject.getInstance().addByName("Verifi", Arrays.asList(new DataNote[]{
                new DataNote("Serial #"),
                new DataNote("IMEI #"),
                new DataNote("Other")
        }));
        TableCollectionNoteProject.getInstance().addByName("Other", Arrays.asList(new DataNote[]{
                new DataNote("Serial #"),
                new DataNote("IMEI #"),
                new DataNote("Other")
        }));
    }

}

File: appMain/src/androidTest/java/com/cartlc/tracker/amock/BootstrapData.kt
/**
 * Copyright 2018, FleetTLC. All rights reserved
 */
package com.cartlc.tracker.amock

import com.cartlc.tracker.model.data.DataAddress
import com.cartlc.tracker.model.table.DatabaseTable

import javax.inject.Inject

/**
 * Created by dug on 5/9/17.
 */

class BootstrapData {

    internal val ADDRESSES = arrayOf(
            DataAddress("Alamo Concrete"),
            DataAddress("CW Roberts"),
            DataAddress("Central Concrete"),
            DataAddress("IMI"),
            DataAddress("Ozinga"),
            DataAddress("Wingra Ready Mix"),
            DataAddress("Point Ready Mix"))

    @Inject
    lateinit var db: DatabaseTable

//    fun init() {
//        DaggerDatabaseTableComponent.builder().build().inject(this)
//        if (db.tableAddress.count() == 0) {
//            db.tableAddress.add(Arrays.asList(*ADDRESSES))
//        }
//        if (SqlTableProjects.instance.count() == 0) {
//            AddCollections()
//            AddNotes()
//        }
//    }
//
//    internal fun AddCollections() {
//        SqlTableCollectionEquipmentProject.instance.addByName("Five Cubits", Arrays.asList(*arrayOf("OBC", "OBC Bracket", "RDT", "VMX", "Tablet", "Charging Converter", "6 pin Canbus Cable", "9 pin Canbus Cable", "Green Canbus Cable", "ODB II Canbus Cable", "GDS Cup", "External Antenna", "Internal Antenna", "External Speaker", "Microphone", "Tablet", "Repair Work", "Speaker Box", "Other")))
//        SqlTableCollectionEquipmentProject.instance.addByName("Digital Fleet", Arrays.asList(*arrayOf("Antenna", "Tablet", "Modem", "JBox", "Canbus", "Ram Mount/Cradle", "Charging Converter", "Repair Work", "Uninstall", "Other")))
//        SqlTableCollectionEquipmentProject.instance.addByName("Smart Witness", Arrays.asList(*arrayOf("KP1S", "CPI", "SVC 1080", "Modem", "Driver Facing Camera", "Back Up Camera", "Side Camera 1", "Side Camera 2", "Other")))
//        SqlTableCollectionEquipmentProject.instance.addByName("Fed Ex", Arrays.asList(*arrayOf("KP1S", "SVA30", "Modem", "Mobileye", "Backup Sensors", "Other")))
//        SqlTableCollectionEquipmentProject.instance.addByName("Verifi", Arrays.asList(*arrayOf("Other")))
//        SqlTableProjects.instance.addTest(TBApplication.OTHER)
//    }
//
//    internal fun AddNotes() {
//        SqlTableCollectionNoteProject.instance.addByName("Five Cubits", Arrays.asList(*arrayOf(DataNote("Serial #", DataNote.Type.ALPHANUMERIC), DataNote("IMEI #", DataNote.Type.NUMERIC), DataNote("Other", DataNote.Type.MULTILINE))))
//        SqlTableCollectionNoteProject.instance.addByName("Digital Fleet", Arrays.asList(*arrayOf(DataNote("Serial #"), DataNote("IMEI #"), DataNote("Other"))))
//        SqlTableCollectionNoteProject.instance.addByName("Smart Witness", Arrays.asList(*arrayOf(DataNote("Serial #"), DataNote("IMEI #"), DataNote("Sim #", DataNote.Type.NUMERIC_WITH_SPACES), DataNote("DRID #", DataNote.Type.TEXT), DataNote("Other"))))
//        SqlTableCollectionNoteProject.instance.addByName("Fed Ex", Arrays.asList(*arrayOf(DataNote("Serial #"), DataNote("IMEI #"), DataNote("Sim #"), DataNote("DRID #"), DataNote("Mobileye", DataNote.Type.TEXT), DataNote("Other"))))
//        SqlTableCollectionNoteProject.instance.addByName("Verifi", Arrays.asList(*arrayOf(DataNote("Serial #"), DataNote("IMEI #"), DataNote("Other"))))
//        SqlTableCollectionNoteProject.instance.addByName("Other", Arrays.asList(*arrayOf(DataNote("Serial #"), DataNote("IMEI #"), DataNote("Other"))))
//    }

}
